account_asset_disposal/models/account_asset_asset.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# Copyright 2017 Luis M. Ontalba - <luis.martinez@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountAssetAsset(models.Model):
+    _inherit = "account.asset.asset"
+
+    state = fields.Selection(
+        selection_add=[('disposed', 'Disposed')],
+    )
+    disposal_date = fields.Date(string="Disposal date")
+    disposal_move_id = fields.Many2one(
+        comodel_name='account.move', string="Disposal move")
+
+    def get_disposal_date(self):
+        return fields.Date.context_today(self)
+
+    @api.multi
+    def set_to_close(self):
+        res = super(AccountAssetAsset, self).set_to_close()
+        if res:
+            self.disposal_move_id = res['res_id']
+        self.write({
+            'state': 'disposed',
+            'disposal_date': self.get_disposal_date(),
+            })
+        return res
+
+    @api.multi
+    def action_disposal_undo(self):
+        for asset in self.with_context(asset_disposal_undo=True):
+            if asset.disposal_move_id:
+                asset.disposal_move_id.button_cancel()
+                asset.disposal_move_id.unlink()
+            if asset.currency_id.is_zero(asset.value_residual):
+                asset.state = 'close'
+            else:
+                asset.state = 'open'
+                asset.method_number = asset.category_id.method_number
```

Comment:
Why assign number of depreciations instead of calling `compute_depreciation_board`? Take into account that assets can be depreciated other way (in Spain with l10n_es_account_asset, with a percentage), so this won't work with that method.